### PR TITLE
Use all the screen for the camera by default and fix aspect ratio in preview

### PR DESCRIPTION
**Pull request by bryant1410**

### CODE REVIEW

File: library/src/main/java/com/afollestad/materialcamera/internal/Camera2Fragment.java [hunk 1]
```diff
@@ -275,23 +275,48 @@ public static Camera2Fragment newInstance() {
...
-        if (size.getWidth() == size.getHeight() * ci.videoPreferredAspect()) {
-          return size;
-        }
-        if (ci.videoPreferredHeight() >= size.getHeight()) {
-          backupSize = size;
-        }
+  public static Size getSizeWithClosestRatio(Size[] sizes, int targetWidth, int targetHeight) {
+    if (sizes == null) return null;
+
+    double MIN_TOLERANCE = 100;
```
Comment (mirland) [inline library/src/main/java/com/afollestad/materialcamera/internal/Camera2Fragment.java]: constant

Comment (mirland) [reply library/src/main/java/com/afollestad/materialcamera/internal/Camera2Fragment.java]: unit?

File: library/src/main/java/com/afollestad/materialcamera/internal/Camera2Fragment.java [hunk 2]
```diff
@@ -275,23 +275,48 @@ public static Camera2Fragment newInstance() {
...
+    double targetRatio = (double) targetHeight / targetWidth;
+    Size optimalSize = null;
+    double minDiff = Double.MAX_VALUE;
+
+    for (Size size : sizes) {
+
+      double ratio = (double) size.getHeight() / size.getWidth();
+
```
Comment (mirland) [inline library/src/main/java/com/afollestad/materialcamera/internal/Camera2Fragment.java]: you can remove it

File: library/src/main/java/com/afollestad/materialcamera/internal/Camera2Fragment.java [hunk 3]
```diff
@@ -275,23 +275,48 @@ public static Camera2Fragment newInstance() {
...
+      if (Math.abs(ratio - targetRatio) < MIN_TOLERANCE) {
+        MIN_TOLERANCE = Math.abs(ratio - targetRatio);
+        minDiff = Double.MAX_VALUE;
+      }
+      else continue;
```
Comment (mirland) [inline library/src/main/java/com/afollestad/materialcamera/internal/Camera2Fragment.java]: `{}`

File: library/src/main/res/layout/mcam_include_camera_preview.xml [hunk 4]
```diff
@@ -2,7 +2,7 @@
 <com.afollestad.materialcamera.internal.AutoFitTextureView
     xmlns:android="http://schemas.android.com/apk/res/android"
     android:id="@+id/texture"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
+    android:layout_width="wrap_content"
```
Comment (mirland) [inline library/src/main/res/layout/mcam_include_camera_preview.xml]: I think it's not needed, you included it, so these params are not important

**Review by mirland — APPROVED** [review]

File: library/src/main/java/com/afollestad/materialcamera/internal/Camera2Fragment.java (hunk 1 above)
Comment (bryant1410) [reply library/src/main/java/com/afollestad/materialcamera/internal/Camera2Fragment.java]: the unit is the same as the rest of the variables, so I don't think is necessary

**Pull request merged**